File: Python/Warmup-1/pos_neg.py
"""

Given 2 int values, return True if one is negative and one is positive. Except if the parameter "negative" is True, then return True only if both are negative.


pos_neg(1, -1, False) → True
pos_neg(-1, 1, False) → True
pos_neg(-4, -5, True) → True

@author unobatbayar
"""
def pos_neg(a, b, negative):
  if negative == True:
    if a < 0 and b < 0:
      return True
  if negative == False:
    if a < 0 and b > 0 or a > 0 and b < 0:
      return True
  return False

File: Java/Warmup-1/posNeg.java
/*
Given 2 int values, return true if one is negative and one is positive. Except if the parameter "negative" is true, then return true only if both are negative.


posNeg(1, -1, false) → true
posNeg(-1, 1, false) → true
posNeg(-4, -5, true) → true
*/

class posNeg{
    public boolean posNeg(int a, int b, boolean negative) {
        if (negative) return (a < 0 && b < 0);
        else {
          return ((a < 0 && b > 0) || (a > 0 && b < 0));
        }
      }      
}